samples/DevTestLabs/AzureFunctions/PowerShell/ApplyWindowsUpdateArtifact.ps1

```diff
@@ -0,0 +1,141 @@
+if (-not $Request.Params.SUBSCRIPTIONID) {
+    Write-Error "Missing [SUBSCRIPTIONID] in the Rest API URL"
+}
+if (-not $Request.Params.RESOURCEGROUPNAME) {
+    Write-Error "Missing [RESOURCEGROUPNAME] in the Rest API URL"
+}
+if (-not $Request.Params.LABNAME) {
+    Write-Error "Missing [LABNAME] in the Rest API URL"
+}
+if (-not $Request.Params.VIRTUALMACHINENAME) {
+    Write-Error "Missing [VIRTUALMACHINENAME] in the Rest API URL"
+}
+if (-not $env:ServicePrincipal_AppId) {
+    Write-Error "Missing [ServicePrincipal_AppId] in the ApplicationSettings for the Azure Function"
+}
+if (-not $env:ServicePrincipal_Key) {
+    Write-Error "Missing [ServicePrincipal_Key] in the ApplicationSettings for the Azure Function"
+}
+if (-not $env:ServicePrincipal_Tenant) {
+    Write-Error "Missing [ServicePrincipal_Tenant] in the ApplicationSettings for the Azure Function"
+}
+
+$javascriptToCloseBrowserTab = @'
+<script language='javascript' type='text/javascript'>
+    // script to close the tab after the page has loaded
+    setTimeout(function() {
+        var x = confirm('Are you sure you want to close this tab?');
+        if (x) {
+            window.open('', '_parent', '');
+            window.close();
+        }
+    }, 5000); // 5 seconds
+</script>
+'@
+
+$htmlPageTemplateResponse = @'
+<html>
+  <head>
+    <title>DevTest Lab Extensions</title>
+    <link rel='icon shortcut' href='https://azurecomcdn.azureedge.net/cvt-cc7cfb2db134841032aa0a589fb58090be0b25d70fb9a887f22a0d84adf021a9/images/icon/favicon.ico'>
+    {JAVASCRIPT}
+  </head>
+  <body>
+    <div>
+        <img alt='LabSuccessfullyUpdated.png' src='{IMAGEURL}'>
+    </div>
+  </body>
+</html>
+'@
+
+# ---------------------------------------------------
+# Setup creds and log into Azure
+# ---------------------------------------------------
+$secpasswd = ConvertTo-SecureString $env:ServicePrincipal_Key -AsPlainText -Force
+$pscreds = New-Object System.Management.Automation.PSCredential ($env:ServicePrincipal_AppId, $secpasswd)
+
+# Log into Azure with the service principal
+$login = Login-AzAccount -ServicePrincipal -Tenant $env:ServicePrincipal_Tenant -Credential $pscreds
+
+# Make sure we're looking at the right subscription
+$sub = Select-AzSubscription -SubscriptionId $Request.Params.SUBSCRIPTIONID
+
+# ---------------------------------------------------
+# Confirm subscription and lab exist
+# ---------------------------------------------------
+
+if ($sub) {
+    # Get the Virtual Machine details
+    try {
+        $vm = Get-AzResource -ResourceGroupName $Request.Params.RESOURCEGROUPNAME -ResourceType 'Microsoft.DevTestLab/labs/virtualmachines' -Name "$($Request.Params.LABNAME)/$($Request.Params.VIRTUALMACHINENAME)" -ODataQuery '$expand=Properties($expand=ComputeVM)' -ApiVersion 2016-05-15
+    }
+    catch {
+        Write-Error "Unable to find the Virtual Machine '$($Request.Params.VIRTUALMACHINENAME)' in Lab '$($Request.Params.LABNAME)'"
```

Comment:
Ok, thanks Leo!  If you want me to change all the write-error to throw, just let me know, otherwise I'll leave it as-is..   Thanks!

Reply:
Leave as is. Just making sure you're aware, in case someone decides to remove the configuration line, and you try to figure out why things continue to run after failure.